### PR TITLE
[MM-47222] add info for enduser with no account and when on closed server

### PR DESCRIPTION
**Pull request by mhd-sln**
#### Summary
add info for enduser on /login page with no user account and when on closed server. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-47222

#### Screenshots
|  before  |  after  |
|----|----|
| Screenshot 2022-10-23 at 19 34 20 | Screenshot 2022-10-28 at 23 38 49 |

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by JulienTant — APPROVED** [review]
LGTM @salmanmanekia!
nit: It would be nice to have an e2e test running this new scenario.

File: components/access_problem/access_problem.scss
Comment (pvev) [inline components/access_problem/access_problem.scss:6]: it is a good practice to nest elements inside a block

```
.AccessProblem {
    &__containter{...}
    &__body{...}
}
```

File: components/access_problem/index.tsx
Comment (pvev) [inline components/access_problem/index.tsx:19]: I think the outer div.AccessProblem and child div.AccessProblem__container are not necessary, you would only probably need the div.AccessProblem__body to work as a wrapper and that is it

File: components/header_footer_route/content_layouts/alternate_link.tsx
Comment (pvev) [inline components/header_footer_route/content_layouts/alternate_link.tsx:12]: it is a good practice to avoid as much as possible the use of `any`, you can define the type of the function like

`onClick?: () => void `(or if it will return an object, the object type)

**Review by pvev — CHANGES_REQUESTED** [review]
looking god so far, some small changes required before approval. Thanks @salmanmanekia

**Comment by salmanmanekia** [comment]
Thanks @pvev and @JulienTant. I have added the test case and implemented the suggestions. Can you please review it once more ?
I would like to highlight that I have added a new test file for [login_close_server_spec.js](https://github.com/mattermost/mattermost-webapp/pull/11474/files#diff-d4546e6891c9ab8c4b94083e1c24889d6f74b233b0eba759f3c5dd59a7673c2e). I tried adding the test case in the already present file `login_spec.js` but applying different server config for different test cases in the same file did not seem to work. Please suggest if you think this could be done in a better way ?

File: e2e/cypress/tests/integration/signin_authentication/login_close_server_spec.js
Comment (JulienTant) [inline e2e/cypress/tests/integration/signin_authentication/login_close_server_spec.js:30]: (not a request for change)
In e2e tests, we try to have assertions that a regular human could do.
In this case, I would probably ask a human to see if they indeed have the 'Don't have an account?' text but I would certainly have them click on the link and make sure they end up on the page that contains "To access your team's workspace, contact your workspace admin". Doing so we validate that the link exists and as a bonus, we make sure the next page is also not broken.
Sometimes we don't have a choice and have to assert on "technical" parts, but whenever possible it's better not to in e2e tests.
Does that make sense?

Comment (salmanmanekia) [reply e2e/cypress/tests/integration/signin_authentication/login_close_server_spec.js:30]: it does. I will improve this.

**Comment by JulienTant** [comment]
> Please suggest if you think this could be done in a better way

You would need to change the values inside of your test case but make sure the `before` or `beforeEach` reset everything back to what is expected by the other test cases.

**Comment by salmanmanekia** [comment]
> > Please suggest if you think this could be done in a better way
> 
> You would need to change the values inside of your test case but make sure the `before` or `beforeEach` reset everything back to what is expected by the other test cases.

I tried but it just won't work so I thought of having a separate file for it

**Review by pvev — APPROVED** [review]
great work @salmanmanekia , thanks! :shipit:

**Review by annelieseklein — APPROVED** [review]

**Review by stevemudie — APPROVED** [review]
I'll test this post merge @mhd-sln so it's ready to go as is now.

**Pull request merged**
